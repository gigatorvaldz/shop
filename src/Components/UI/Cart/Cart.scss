.cart {
    display: flex;
    &__logo {
        position: relative;
        width: 46px;
        height: 46px;
        background: no-repeat center/80% url("../../../img/cart-icon.svg");

        & span {
            right: -5px;
            background: #ffc85e;
            border: 3px solid #ffffff;
            width: 25px;
            height: 25px;
            display: flex;
            justify-content: center;
            align-items: center;
            border-radius: 50%;
            text-align: center;
            color: #fff;
            font-weight: 700;
            font-size: 14px;
            position: absolute;
        }
    }

    &__info {
        margin-left: 15px;
        display: flex;
        flex-direction: column;

        &-name {
            font-weight: 300;
            font-size: 12px;
            line-height: 150%;
            color: #3f4e65;
        }

        &-price {
            font-weight: 600;
            font-size: 14px;
            line-height: 150%;
            color: #111111;
        }
    }
}
