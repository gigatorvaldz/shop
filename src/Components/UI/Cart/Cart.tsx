import React from "react";
import "./Cart.scss";
import classNames from "classnames";

interface CartPropsI {
  total: number;
  price?: number;
}

function Cart({ total, price = -1 }: CartPropsI) {
  return (
    <div data-testid="cart-test-id" className="lower-header__cart cart">
      <div className="cart__logo">
        <span>{total}</span>
      </div>
      <div
        className={classNames(
          {
            "visually-hidden": price < 0,
          },
          "cart__info"
        )}
      >
        <p className="cart__info-name">Корзина</p>
        <p className="cart__info-price">{price} ₸</p>
      </div>
    </div>
  );
}

export default Cart;
