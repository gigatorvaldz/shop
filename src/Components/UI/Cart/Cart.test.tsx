import { render, screen } from "@testing-library/react";
import Cart from "./Cart";

test("Cart component snapshot test", () => {
  render(<Cart total={123} price={12400} />);
  let cart = screen.getByTestId("cart-test-id");

  expect(cart).toMatchSnapshot();
});

test("Cart component input test", () => {
  render(<Cart total={123321} price={12400} />);
  let total = screen.getByText("123321");
  let price = screen.getByText("12400 â‚¸");

  expect(total).toBeInTheDocument;
  expect(price).toBeInTheDocument;
});
