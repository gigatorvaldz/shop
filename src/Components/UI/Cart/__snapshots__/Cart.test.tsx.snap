// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component snapshot test 1`] = `
<div
  class="lower-header__cart cart"
  data-testid="cart-test-id"
>
  <div
    class="cart__logo"
  >
    <span>
      123
    </span>
  </div>
  <div
    class="cart__info"
  >
    <p
      class="cart__info-name"
    >
      Корзина
    </p>
    <p
      class="cart__info-price"
    >
      12400
       ₸
    </p>
  </div>
</div>
`;
